src/Compilers/CSharp/Portable/Symbols/Symbol.cs

```diff
@@ -346,6 +346,9 @@ internal virtual LexicalSortKey GetLexicalSortKey()
         public Location GetFirstLocation()
             => TryGetFirstLocation() ?? throw new InvalidOperationException("Symbol has no locations");
 
+        public Location GetFirstLocationOrNone()
```

Comment:
💭 It seems like this could be an extension method since `TryGetFirstLocation` already exists and this is basically doing the same thing.